flame-1.9.1/lib/src/game/flame_game.dart:279:30: Error: Member not found: 'hidden'.

<!-- When reporting a bug, please read this complete template and fill all the questions in order to get a better response -->
This is the first time I am working with the Flame framework, and following the tutorial provided on
[Bare Flame game](https://docs.flame-engine.org/latest/tutorials/bare_flame_game.html), I have this code.

`import 'package:flame/game.dart';
import 'package:flutter/widgets.dart';

void main() {
  final game = FlameGame();
  runApp(GameWidget(game: game));
}`

# Current bug behavior
<!-- What is the current behavior that you see? -->
This is the behaviour of the program when I run it.
(M2101K7BI is my physical device's name)

`Launching lib\main.dart on M2101K7BI in debug mode...
Running Gradle task 'assembleDebug'...
../../../../AppData/Local/Pub/Cache/hosted/pub.dev/flame-1.9.1/lib/src/game/flame_game.dart:279:30: Error: Member not found: 'hidden'.
      case AppLifecycleState.hidden:
                             ^^^^^^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* Where:
Script 'C:\Users\user\Flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1201

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\Users\user\Flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 18s
Exception: Gradle task assembleDebug failed with exit code 1
`

# Steps to reproduce
<!-- This one is very important, please be very precise in how we can reproduce this bug -->
<!-- If possible please report steps based on the example from this plugin! -->
<!-- If you can make a minimal reproducible example it is incredibly helpful, the simplest way is to share a link from https://zapp.run, you can start from https://zapp.run/edit/flame where all dependencies are already set up. -->

This error occurred the first time I ran the program, so I don't exactly know the procedure to reproduce the same.

# Flutter doctor output
<!-- Execute in a terminal and put output into code block below -->
```
[√] Flutter (Channel stable, 3.10.6, on Microsoft Windows [Version 10.0.19045.3324], locale en-IN)
    • Flutter version 3.10.6 on channel stable at C:\Users\user\Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (3 months ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Program Files\Android\Sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[!] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32210.238
    X The current Visual Studio installation is incomplete. Please reinstall Visual Studio.

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code, 64-bit edition (version 1.81.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.56.0

[√] Connected device (4 available)
    • M2101K7BI (mobile) • IRGEH6ZLDUTGT86H • android-arm64  • Android 12 (API 31)
    • Windows (desktop)  • windows          • windows-x64    • Microsoft Windows [Version 10.0.19045.3324]
    • Chrome (web)       • chrome           • web-javascript • Google Chrome 117.0.5938.92
    • Edge (web)         • edge             • web-javascript • Microsoft Edge 117.0.2045.41

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

# More environment information

* Flame flame: ^1.9.1
* Platform affected: android
* Platform version affected: android 12

--- a/.github/workflows/cicd.yml
+++ b/.github/workflows/cicd.yml
@@ -14,11 +14,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: subosito/flutter-action@v2
-        with:
-          flutter-version: '3.13.2'
-          channel: 'stable'
-          cache: true
-      - uses: bluefireteam/melos-action@main
+      - uses: bluefireteam/melos-action@v2
       - run: melos run format-check
 
   analyze:
@@ -27,10 +23,20 @@ jobs:
       - uses: actions/checkout@v3
       - uses: subosito/flutter-action@v2
         with:
-          flutter-version: '3.13.2'
-          channel: 'stable'
+          flutter-version: '3.13.0'
       - uses: bluefireteam/melos-action@v2
-      - name: "Analyze"
+      - name: "Analyze with lowest supported version"
+        uses: invertase/github-action-dart-analyzer@v2.0.0
+        with:
+          fatal-infos: true
+
+  analyze-latest:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: subosito/flutter-action@v2
+      - uses: bluefireteam/melos-action@v2
+      - name: "Analyze with latest stable"
         uses: invertase/github-action-dart-analyzer@v2.0.0
         with:
           fatal-infos: true
@@ -50,11 +56,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: subosito/flutter-action@v2
-        with:
-          flutter-version: '3.13.2'
-          channel: 'stable'
       - uses: bluefireteam/melos-action@v2
-
       - name: Install DCM
         uses: CQLabs/setup-dcm@v1
         with:
@@ -71,10 +73,8 @@ jobs:
       - uses: actions/checkout@v3
       - uses: subosito/flutter-action@v2
         with:
-          flutter-version: '3.13.2'
-          channel: 'stable'
           cache: true
-      - uses: bluefireteam/melos-action@main
+      - uses: bluefireteam/melos-action@v2
       - name: Run tests
         run: melos test
   # END TESTING STAGE

--- a/doc/flame/examples/pubspec.yaml
+++ b/doc/flame/examples/pubspec.yaml
@@ -5,7 +5,7 @@ publish_to: none
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/doc/tutorials/bare_flame_game.md
+++ b/doc/tutorials/bare_flame_game.md
@@ -3,7 +3,7 @@
 This tutorial assumes that you have basic familiarity with using the command line, and the following
 programs on your computer (all of them are free):
 
-- [Flutter], version 3.3.0 or above.
+- [Flutter], version 3.13.0 or above.
 - [Android Studio], or any other IDE for example [Visual Studio Code].
 - [git] (optional), to save your project on GitHub.
 
@@ -25,7 +25,7 @@ $ flutter doctor
 ```
 
 Your output will be slightly different, but the important thing is to verify that no errors are
-reported and that your Flutter version is at least **3.3.0**.
+reported and that your Flutter version is at least **3.13.0**.
 
 
 ## 2. Create the Project Directory

--- a/examples/games/rogue_shooter/pubspec.yaml
+++ b/examples/games/rogue_shooter/pubspec.yaml
@@ -7,7 +7,7 @@ version: 0.1.0
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/examples/games/trex/pubspec.yaml
+++ b/examples/games/trex/pubspec.yaml
@@ -7,7 +7,7 @@ version: 0.1.0
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   collection: ^1.16.0

--- a/examples/pubspec.yaml
+++ b/examples/pubspec.yaml
@@ -7,7 +7,7 @@ version: 0.1.0
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   dashbook: ^0.1.12

--- a/packages/flame/example/pubspec.yaml
+++ b/packages/flame/example/pubspec.yaml
@@ -5,7 +5,7 @@ publish_to: 'none'
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/packages/flame/pubspec.yaml
+++ b/packages/flame/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   collection: ^1.17.1

--- a/packages/flame_audio/pubspec.yaml
+++ b/packages/flame_audio/pubspec.yaml
@@ -10,7 +10,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   audioplayers: ^5.0.0

--- a/packages/flame_fire_atlas/pubspec.yaml
+++ b/packages/flame_fire_atlas/pubspec.yaml
@@ -10,7 +10,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   archive: ^3.3.9

--- a/packages/flame_forge2d/example/pubspec.yaml
+++ b/packages/flame_forge2d/example/pubspec.yaml
@@ -7,7 +7,7 @@ version: 1.0.0+1
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   dashbook: ^0.1.12

--- a/packages/flame_forge2d/pubspec.yaml
+++ b/packages/flame_forge2d/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/packages/flame_jenny/pubspec.yaml
+++ b/packages/flame_jenny/pubspec.yaml
@@ -10,7 +10,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   collection: ^1.17.1

--- a/packages/flame_lottie/pubspec.yaml
+++ b/packages/flame_lottie/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: '>=3.3.0'
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/packages/flame_markdown/pubspec.yaml
+++ b/packages/flame_markdown/pubspec.yaml
@@ -10,7 +10,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/packages/flame_noise/pubspec.yaml
+++ b/packages/flame_noise/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   fast_noise: ^1.0.1

--- a/packages/flame_oxygen/pubspec.yaml
+++ b/packages/flame_oxygen/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/packages/flame_rive/pubspec.yaml
+++ b/packages/flame_rive/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/packages/flame_spine/pubspec.yaml
+++ b/packages/flame_spine/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   collection: ^1.17.1

--- a/packages/flame_test/pubspec.yaml
+++ b/packages/flame_test/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   flame: ^1.9.1

--- a/packages/flame_tiled/pubspec.yaml
+++ b/packages/flame_tiled/pubspec.yaml
@@ -9,7 +9,7 @@ funding:
 
 environment:
   sdk: ">=3.0.0 <4.0.0"
-  flutter: ">=3.3.0"
+  flutter: ">=3.13.0"
 
 dependencies:
   collection: ^1.17.1

